proposals/multi-cluster-config.md

```diff
@@ -0,0 +1,155 @@
+<!--
+Copyright 2015-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You may
+not use this file except in compliance with the License. A copy of the
+License is located at
+
+http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+-->
+
+# Introduction
+ECS CLI currently only support one cluster configuration which results in a number of issues for the users:
+1. When customer switch clusters or regions within the same account, they need to reconfigure the setup.
+1. When customer is running two Jenkins builds simultaneously, the config file gets reconfigured while one job is still processing.
+1. When customers work as a team, each person has to run `ecs-cli configure` to configure the correct cluster and region.
+1. Everything (cluster name, region, creds) should be configurable via arguments or environment variables.
+
+# Solution
+When users run `ecs-cli configure`, ECS CLI will configure their cluster configuration in `~/.ecs/ecs.config`
+```yaml
+default: beta
+clusters:
+  beta:
+  - cluster: cli-demo
+  - region: us-west-2
+  gamma:
+  - cluster: cli-demo2
+  - region: us-west-1
+  prod:
+  - ...
+```
+
+When users run `ecs-cli configure profile`, ECS CLI will configure their credentials in `~/.ecs/ecs.profile`
+```yaml
+default: my_profile
+ecs_profiles:
+  my_profile:
+  - aws_access_key_id     : **********
+  - aws_secret_access_key : **********
+  dev_profile:
+  - aws_access_key_id     : **********
+  - aws_secret_access_key : **********
+  team_profile:
+  - ...
+```
+
+Other commands will have additional flags
+* **--cluster-config**
+* --cluster
+* --region
+* --aws-profile (this is --profile today)
+* **--ecs-profile**
+
+# Workflow
+### Single cluster first run
+1. Configure ECS credentials (Stores in `~/.ecs/ecs.profile`)
+   ```
+   ecs-cli configure  profile --access-key ********** --secret-key ********** --profile-name my_profile
+   ```
+1. Configure cluster (Stores in `~/.ecs/ecs.config`)
+   ```
+   ecs-cli configure --cluster cli-demo --region us-west-2 --config-name beta_config
+   ```
```

Comment:
We allow multiple cluster configurations in ~/.ecs/ecs.config, `--config-name` is a way to identify the different cluster profile which can be specified at runtime. Assume you have an existing cluster config "ecs" and you would like to configure a different cluster but not overwriting the previous one, you can run the command with `--config-name` with "beta_config" to add the new cluster in the cluster config.
```yaml
default: ecs
clusters:
  ecs: 
      cluster: ecs-test
      region: us-west-1
  beta_config:
      cluster: cli-demo
      region: us-west-2
```